### PR TITLE
RHPAM-2987: Stuner - The leap days are not being validated in timer events

### PR DESCRIPTION
**Pull request by handreyrc**
@romartin , @inodeman: Please, could you review this PR?

### CODE REVIEW

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

File: kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/forms/validation/timerEditor/TimerSettingsValueValidator.java [hunk 1]
```diff
@@ -172,10 +181,36 @@ private static boolean isValidQuartzExpression(final String value) {
...
+    }
+
+    // if the ISO expression has dates, they must be valid ones
+    private static boolean mayHaveValidDate(final String value) {
+        List<String> matches = fetchDateStrings(value);
+        if (!matches.isEmpty()) {
+            try {
+                DateTimeFormat dateTimeFormat = DateTimeFormat.getFormat("yyyy-MM-dd");
+                matches.forEach(dateTimeFormat::parseStrict);
+            } catch (Exception e) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/forms/validation/timerEditor/TimerSettingsValueValidator.java:210]: I understand that the key point is the use of the `DateTimeFormat.parseStrict` method. Once calling it, if an execptoin occurs, we can assume that the given date is not valid. But maybe we can be more specific on the exception type being catched on this assumption - does it makes sense assuming once any "generic" exception occurs is due to the date is not valid (eg: leap days)? Is there any other Exception subtype we could use for the catch block instead? Just brainstorming, just asking :)

Comment (handreyrc) [reply kie-wb-common-stunner/kie-wb-common-stunner-sets/kie-wb-common-stunner-bpmn/kie-wb-common-stunner-bpmn-api/src/main/java/org/kie/workbench/common/stunner/bpmn/forms/validation/timerEditor/TimerSettingsValueValidator.java]: Hey @romartin, The parseStrict throws one one kind of exception "IllegalArgumentException", so I thought it was ok implementing this way. However, looking back at the code I agree it is not clear without checking the API.
I've already changed the code to make this exception catching more specific.
Thanks

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
need for feedback here please @LuboTerifaj

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
@romartin , @LuboTerifaj: Here goes the link to the artifacts for testing this fix.
https://drive.google.com/file/d/1iOK5l90i2Tuouhl3z14azOzHY0Ml0ry5/view?usp=sharing
Thanks

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
Jenkins execute compile downstream build

**Comment by handreyrc** [comment]
Jenkins retest this

**Comment by handreyrc** [comment]
Hey @romartin  we have the builds green. 
Thanks

**Pull request merged**